merkledag/node.go

```diff
@@ -107,9 +110,17 @@ func (n *Node) AddNodeLinkClean(name string, that *Node) error {
 	if err != nil {
 		return err
 	}
-	lnk.Name = name
+	n.AddRawLink(name, lnk)
 
+	return nil
+}
+
+// AddRawLink adds a pre-provided link to this node
+func (n *Node) AddRawLink(name string, lnk *Link) error {
```

Comment:
so this means, we already have a link object and sharing it across objects?

hmmm that may be ok, not sure. @whyrusleeping do you know if this will violate any invariants?

(i'd just dupe the link object, so that each node has complete control over the links... but maybe we should move to a model like you suggest)